### PR TITLE
Solidity version

### PR DESCRIPTION
**Pull request by friedger**
Mainly based on https://github.com/nibbstack/erc721

Added `tokenIdToTierId`

### CODE REVIEW

File: backend-solidity/src/contracts/ryder-nft.sol [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                    dicksonParameter *
+                    _tokenId**3 +
+                    3002 *
+                    dicksonParameter**2 *
+                    _tokenId +
+                    dicksonParameter) % MAX_TOKENS;
+    }
+
+    function tier(uint256 tierId) public pure returns (uint256) {
+        if (tierId == 0) return 0;
```
Comment (friedger) [inline backend-solidity/src/contracts/ryder-nft.sol:102]: Range of tierId is 0 to 5002, therefore, this line needs to be removed.

File: backend-solidity/src/contracts/ryder-nft.sol [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        validNFToken(tokenId)
+        returns (uint256)
+    {
+        return tier(tokenIdToTierId(tokenId));
+    }
+
+    // Admin functions
+
+    function setDicksonParameter(uint256 _dicksonParameter) external adminOnly {
+        dicksonParameter = _dicksonParameter;
```
Comment (friedger) [inline backend-solidity/src/contracts/ryder-nft.sol:127]: This should be guarded by a check for `dicksonParameterSet`

Comment (friedger) [reply backend-solidity/src/contracts/ryder-nft.sol:127]: Should we use `time` here and do mod 5003 when setting the parameter?

**Review by friedger — COMMENTED** [review]
Should we use time as the parameter for the dickson setter function?

File: backend-solidity/src/contracts/ryder-nft.sol (hunk 2 above)
Comment (MarvinJanssen) [reply backend-solidity/src/contracts/ryder-nft.sol:127]: @friedger what is the upside to doing mod there?

Comment (friedger) [reply backend-solidity/src/contracts/ryder-nft.sol:127]: @MarvinJanssen It is easier to verify that we actually used the correct timestamp as promised.

File: backend-stacks/contracts/ryder-nft.clar [hunk 3]
```diff
@@ -144,6 +145,7 @@
 (define-public (set-admin (new-admin principal) (value bool))
   (begin
     (try! (check-is-admin))
+    (asserts! (not (is-eq tx-sender)) err-not-allowed)
```
Comment (friedger) [inline backend-stacks/contracts/ryder-nft.clar:148]: The second parameter is missing.

**Comment by friedger** [comment]
LGTM

**Pull request merged**